### DISCUSSION TITLE
重入攻击中的一个小问题

### DISCUSSION THREAD

**Discussion by RealWillsLi** [AmazingAng/WTF-Solidity#468] [Q&A] [answered]
![$2J4BACX6V~6)KLHF3EDDBP](https://user-images.githubusercontent.com/89158403/230097938-f7482658-89f5-400f-9800-da5cc2efcf65.png)
在fallback函数中，require和if语句有什么区别吗，为什么if可以成功而require不行

**Comment by Big-Aaron** [comment, marked as answer]
require检查不通过，交易就失败了。if检查不通过，就执行else交易不会失败。

**Reply by RealWillsLi** [reply]
多谢了，最开始脑子短路了，后面再一看就看出错误的原因了
